tests/system-tests/internal/supporttools/tcpdump.go

```diff
@@ -0,0 +1,543 @@
+		SeccompProfile: &corev1.SeccompProfile{
+			Type: corev1.SeccompProfileTypeRuntimeDefault,
+		},
+		Capabilities: &corev1.Capabilities{
+			Add: []corev1.Capability{
+				"SETFCAP",
+				"CAP_NET_RAW",
+				"CAP_NET_ADMIN",
+			},
+		},
+	}
+
+	glog.V(100).Infof("Setting SecurityContext")
+
+	deployContainer = deployContainer.WithSecurityContext(securityContext)
+
+	glog.V(100).Infof("Dropping ALL security capability")
+
+	deployContainer = deployContainer.WithDropSecurityCapabilities([]string{"ALL"}, true)
+
+	glog.V(100).Infof("Enable TTY and Stdin; needed for immediate log propagation")
+
+	deployContainer = deployContainer.WithTTY(true).WithStdin(true)
+
+	glog.V(100).Infof("%q container's  definition:\n%#v", cName, deployContainer)
+
+	return deployContainer
+}
+
+// SendTrafficCheckTCPDumpLogs is a wrapper function to reduce code duplication when probing for target IPs.
+func SendTrafficCheckTCPDumpLogs(
+	apiClient *clients.Settings,
+	proberPodObj *pod.Builder,
+	tcpdumpPodNamespace,
+	tcpdumpPodSelectorLabel,
+	targetHost,
+	targetProtocol,
+	targetPort string) error {
+	timeout := 3 * time.Minute
+	err := wait.PollUntilContextTimeout(
+		context.TODO(),
+		3*time.Second,
+		timeout,
+		true,
+		func(ctx context.Context) (bool, error) {
+			glog.V(100).Infof("Verifying that the expected outbound request can be seen in the tcpdump logs")
+
+			result, err := sendProbeAndCheckTCPDumpLogs(
+				apiClient,
+				proberPodObj,
+				tcpdumpPodNamespace,
+				tcpdumpPodSelectorLabel,
+				targetProtocol,
+				targetHost,
+				targetPort)
+
+			if err == nil && result {
+				return true, nil
+			}
+
+			return false, nil
+		})
+
+	if err != nil {
+		return fmt.Errorf("expected request was not found in the packet sniffer logs; %w", err)
+	}
+
+	return nil
+}
+
+// sendProbeAndCheckTCPDumpLogs sends requests from the prober pod.
+func sendProbeAndCheckTCPDumpLogs(
+	apiClient *clients.Settings,
+	proberPod *pod.Builder,
+	tcpdumpPodNamespace,
+	tcpdumpPodSelectorLabel,
+	targetProtocol,
+	targetHost,
+	targetPort string) (bool, error) {
+	timeStart := time.Now()
+
+	glog.V(100).Infof("Sending request with a unique search string from "+
+		"prober pod %s/%s", proberPod.Definition.Namespace, proberPod.Definition.Name)
+
+	searchString, err := sendProbeToHostPort(
+		proberPod,
+		targetProtocol,
+		targetHost,
+		targetPort)
+	if err != nil {
+		return false, err
+	}
+
+	glog.V(100).Infof("request sended from the prober pod %s/%s",
+		proberPod.Definition.Namespace, proberPod.Definition.Name)
+
+	time.Sleep(time.Minute)
+
+	for i := 0; i < logScanMaxTries; i++ {
+		glog.V(100).Infof("Making sure that request with the search string %s were seen (try %d of %d)",
+			searchString, i+1, logScanMaxTries)
+
+		numberFound, _, err := ScanTCPDumpPodLogs(
+			apiClient,
+			proberPod,
+			targetProtocol,
+			tcpdumpPodNamespace,
+			tcpdumpPodSelectorLabel,
+			searchString,
+			timeStart)
+
+		if err != nil {
+			return false, err
```

Comment:
1. please log an error
2. this immediately returns from the method and doesn't retry even if there available retries left, is that expected?

Reply:
fixed